app/src/main/java/com/spoony/spoony/core/designsystem/component/textField/SpoonyIconButtonTextField.kt

```diff
@@ -0,0 +1,120 @@
+package com.spoony.spoony.core.designsystem.component.textField
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.spoony.spoony.R
+import com.spoony.spoony.core.designsystem.theme.SpoonyAndroidTheme
+import com.spoony.spoony.core.util.extension.noRippleClickable
+
+@Composable
+fun SpoonyIconButtonTextField(
+    value: String,
+    onValueChanged: (String) -> Unit,
+    placeholder: String,
+    onDeleteClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    showDeleteIcon: Boolean = true,
+    maxLength: Int = Int.MAX_VALUE,
+    helperText: String? = null
+) {
+    var isFocused by remember { mutableStateOf(false) }
+    val spoonyColors = SpoonyAndroidTheme.colors
+
+    val borderColor = remember(isFocused) {
+        when {
+            isFocused -> spoonyColors.main400
+            else -> spoonyColors.gray100
+        }
+    }
+
+    Column(verticalArrangement = Arrangement.spacedBy(8.dp)) {
+        SpoonyBasicTextField(
+            value = value,
+            placeholder = placeholder,
+            onValueChanged = { newText ->
+                if (newText.length <= maxLength) {
+                    onValueChanged(newText)
+                }
+            },
+            borderColor = borderColor,
+            modifier = modifier,
+            onFocusChanged = { isFocused = it },
+            trailingIcon = {
+                if (showDeleteIcon && value.isNotEmpty()) {
+                    Icon(
+                        painter = painterResource(id = R.drawable.ic_minus_gray400_24),
```

Comment:
p1) painter 내부 코드를 보시면 UnStable 합니다. 그렇기 때문에 리컴포지션의 대상이 돼요.
ImageVector를 사용하신다면 Stable 하기 때문에 불필요한 리컴포지션을 줄일 수 있습니다.
수정 부탁드릴게요 :)
```suggestion
                        imageVector = ImageVector.vectorResource(id = R.drawable.ic_minus_gray400_24),
```